ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -464,7 +464,7 @@ Operator genOPTree(ASTNode ast, PlannerContext plannerCtx) throws SemanticExcept
           // 0. Gen Optimized Plan
           RelNode newPlan = logicalPlan();
 
-          if (this.conf.getBoolVar(HiveConf.ConfVars.HIVE_CBO_RETPATH_HIVEOP)) {
+          if (ReturnPathManager.shouldUse()) {
             if (cboCtx.type == PreCboCtx.Type.VIEW && !materializedView) {
```

Comment:
We should only fail if return path is `true` rather than `supported`?

Reply:
I don't understand this, the line here is to decide if we'd use the "return path way" or the "non return path way". What are your referring to here as failing?

My concept was that in case of "supported" we decide between these two by applying some logic that decides whether the actual command is in the supported set of commands that are already handled properly with return path, or not. If it is in it, then it works the same way as "true", if it is not, then it works the same way as "false".

Reply:
If it is a create view statement, we fail with `Create view is not supported in cbo return path.`. I would expect that since view creation is not supported in cbo return path, we just either 1) do not use return path for view creation, or 2) we only fail if we are setting this to `true`?

Reply:
Ok, I see it now. Is it the plan not to use RP for view creation ever? In this case we must keep the non return path way at the end?
Or is it the case, that eventually we'll use RP for view creation as well? Because in this case we can add the exclusion of view creation to the ReturnPathManager to return false in case of "supported" for now.

Reply:
> Or is it the case, that eventually we'll use RP for view creation as well?
Exactly, this should be the case so we hit same validation path for queries and views.